docs/implementation.html

```diff
@@ -78,7 +78,49 @@ <h5><a id="controlbatch" href="#controlbatch">5.3.1.1 Control Batches</a></h5>
     </pre></p>
     <p>The schema for the value of a control record is dependent on the type. The value is opaque to clients.</p>
 
-	<h4><a id="record" href="#record">5.3.2 Record</a></h4>
+    <h5><a id="messageset" href="#messageset">5.3.1.2 Old Message Format</a></h5>
+    <p>
+        Prior to Kafka 0.11 messages were transferred and stored in a format called MessageSets. In MessageSets each message has its own metadata. Note that although MessageSets are formally arrays,
+        they are not preceded by an int32 array size like other array elements in the protocol.
+    </p>
+
+    <b>Message Set:</b><br>
+    <p><pre class="brush: java;">
+    MessageSet (Version: 0) => [offset message_size message]
+        offset => INT64
+        message_size => INT32
+        message => crc magic_byte attributes key value
+            crc => INT32
+            magic_byte => INT8
+            attributes => INT8
+            key => BYTES
+            value => BYTES
+    </pre></p>
+    <p><pre class="brush: java;">
+    MessageSet (Version: 1) => [offset message_size message]
+        offset => INT64
+        message_size => INT32
+        message => crc magic_byte attributes key value
+            crc => INT32
+            magic_byte => INT8
+            attributes => INT8
+            timestamp =>INT64
+            key => BYTES
+            value => BYTES
+    </pre></p>
+    <p>
+        When the producer is sending non compressed messages, it can set the offsets to anything.
+        When the producer is sending compressed messages, to avoid server side re-compression, each compressed message should have offset starting from 0 and increasing by one for each inner message in the compressed message.
```

Comment:
I think we should emphasize that this capability is only supported in v1. For v0 messages, the inner offsets are not relative. For the purpose of the producer, it does not really matter since offsets always start at 0, but for the consumer the distinction is important.